### PR TITLE
Output particle heating XS in chi-tech format

### PR DESCRIPTION
**Pull request by markNbaHoang12**
Note: The "Remove the testing" commit is just me removing a few bookmarks for the parts that I add in the codes. It doesn't affect the functionality in anyway

### CODE REVIEW

File: example2a.sh [hunk 1]
```diff
@@ -21,7 +21,7 @@ output_file_prefix="O16_n30g12_njoy2021"
 cd njoy_runner || exit
 
 python3 generate_njoy_mgxs.py \
---njoy_exec_name=njoy \
+--njoy_exec_name=njoy21 \
```
Comment (ragusa) [inline example2a.sh:24]: please leave njoy as in the orginal file

File: njoy_processor/Utils_Combiner.py [hunk 2]
```diff
@@ -180,20 +196,16 @@ def BuildCombinedData(raw_njoy_data, plot = False, verbose = False):
         rx_name = "(n,{:01d}n)".format(nn)
         n_to_n_nxn_keys.append(rx_name)
     # Freegas thermal scattering
-    n_to_n_freegas_keys = ["mt221"]
+    n_to_n_freegas_keys = ["free-gas"]
```
Comment (ragusa) [inline njoy_processor/Utils_Combiner.py:199]: you are reverting to an old version of the file. do not do that. I will not be able to allow that merge request.

File: njoy_processor/Utils_Combiner.py [hunk 3]
```diff
@@ -180,20 +196,16 @@ def BuildCombinedData(raw_njoy_data, plot = False, verbose = False):
...
-    # Neutron to neutron other reactions
-    n_to_n_other_keys = ["(n,np)", "(n,nd)", "(n,na)",
-                         "(n,n2a)", "(n,n3a)"]
     # Neutron to gamma reactions
-    n_to_g_transfer_keys = ["(n,g)", "(n,inel)",
-                            "(n,np)", "(n,nd)",
-                            "(n,p)", "(n,d)", "(n,t)", "(n,a)", "(n,2a)",  
-                            "(n,na)", "(n,n2a)", "(n,n3a)",
-                            "(n,nc)"]
+    n_to_g_transfer_keys = ["(n,g)", "(n,inel)", "(n,np)",
```
Comment (ragusa) [inline njoy_processor/Utils_Combiner.py]: you are reverting to an old version of the file. do not do that. I will not be able to allow that merge request.

**Review by ragusa — CHANGES_REQUESTED** [review]
several times in combiner, you are reverting to an old version of the file. do not do that. I will not be able to allow that merge request.

File: njoy_processor/Utils_Combiner.py (hunk 3 above)
Comment (ragusa) [reply njoy_processor/Utils_Combiner.py]: you are missing n_to_n_other_keys which is currently in my file. are you sure your branch is based on the latest version of my repo? you have have to do a pull from upstream (google this) to incorporate my changes. A fetch does get my changes in your local database but does not apply them. Maybe fetch from upstream + pull is what you need.

File: njoy_processor/Utils_Combiner.py [hunk 4]
```diff
@@ -180,20 +196,16 @@ def BuildCombinedData(raw_njoy_data, plot = False, verbose = False):
...
+                            "(n,nd)", "(n,p)", "(n,d)",
+                            "(n,t)", "(n,a)",
```
Comment (ragusa) [inline njoy_processor/Utils_Combiner.py]: smae here

File: njoy_processor/Utils_Combiner.py [hunk 5]
```diff
@@ -243,13 +255,6 @@ def BuildCombinedData(raw_njoy_data, plot = False, verbose = False):
...
             with_sab = True
 
-    # Adding all the other neutron to neutron data
-    nranges_to_nranges_other = []
-    for rxn in n_to_n_other_keys:
-        if rxn in transfer_matrices['neutron']:
-            mat = transfer_matrices['neutron'][rxn]
-            nranges_to_nranges_other.append(mat)
-
     # Adding all the neutron to gamma data
```
Comment (ragusa) [inline njoy_processor/Utils_Combiner.py]: missing "# Adding all the other neutron to neutron data" and below

File: njoy_processor/Utils_Combiner.py [hunk 6]
```diff
@@ -279,9 +284,6 @@ def BuildCombinedData(raw_njoy_data, plot = False, verbose = False):
     for range_data in nranges_to_nranges_sab_inel:
         if range_data:
             max_num_moms = max(max_num_moms, len(range_data[0]) - 2)
-    for range_data in nranges_to_nranges_other:
-        if range_data:
-            max_num_moms = max(max_num_moms, len(range_data[0]) - 2)
     for range_data in nranges_to_granges:
```
Comment (ragusa) [inline njoy_processor/Utils_Combiner.py]: something is missing

File: njoy_processor/Utils_Combiner.py [hunk 7]
```diff
@@ -330,7 +332,6 @@ def AddTransferGamma(data_values):
     transfer_sab_el = np.copy(transfer_mats)
     transfer_sab_inel = np.copy(transfer_mats)
     transfer_freegas = np.copy(transfer_mats)
-    transfer_n_other = np.copy(transfer_mats)
 
     # Regular elastic scatter (MT-2)
```
Comment (ragusa) [inline njoy_processor/Utils_Combiner.py]: missing

File: njoy_processor/Utils_Combiner.py [hunk 8]
```diff
@@ -368,19 +369,12 @@ def AddTransferGamma(data_values):
         AddTransferNeutron(range_data)
     transfer_sab_inel = np.copy(transfer_mats)
```
Comment (ragusa) [inline njoy_processor/Utils_Combiner.py]: missing

File: njoy_processor/Utils_Combiner.py [hunk 9]
```diff
@@ -368,19 +369,12 @@ def AddTransferGamma(data_values):
...
-        AddTransferNeutron(range_data)
-    transfer_n_other = np.copy(transfer_mats)
-
     # Build base transfer matrices from regular scattering processes
     transfer_mats = [0.0 * mat for mat in transfer_mats]
     for m in range(0, max_num_moms):
         transfer_mats[m] += transfer_el[m]
         transfer_mats[m] += transfer_inel[m]
         transfer_mats[m] += transfer_nxn[m]
-        transfer_mats[m] += transfer_n_other[m]
```
Comment (ragusa) [inline njoy_processor/Utils_Combiner.py]: missing

File: njoy_processor/Utils_Combiner.py [hunk 10]
```diff
@@ -429,6 +412,14 @@ def AddTransferGamma(data_values):
     for range_data in granges_to_granges:
         AddTransferGamma(range_data)
```
Comment (ragusa) [inline njoy_processor/Utils_Combiner.py]: order of operations not the same. you have an old version

**Review by ragusa — CHANGES_REQUESTED** [review]
see if fetch + pull gets you my latest version of the file

**Comment by markNbaHoang12** [comment]
Dear Dr. Ragusa,

I'm sorry for the late reply. I looked into your comments on my pull
request and I've checked out if I could fetch+pull to get the latest
version of your repository. But it seems like from both the github website
and from the command line on the cluster, my version is currently up to
date with yours, apart from my addition that you weren't able to merge. I
attached the screenshot below for your reference, I've also tried to
manually compare the two Utils_Combiner, but they seem to be the same.
Should I reclone into the cluster and redo the pull request to make sure
everything is exactly the same ?

Mark Nguyen
[image: image.png]

Vào Th 7, 26 thg 6, 2021 vào lúc 10:33 Jean Ragusa <
***@***.***> đã viết:

**Review by ragusa — APPROVED** [review]

**Pull request merged**